pkg/install/install.go

```diff
@@ -29,17 +30,19 @@ func Kourier() error {
 	fmt.Println("Starting Networking layer install...")
 
 	kourier := exec.Command("kubectl", "apply", "-f", "https://github.com/knative-sandbox/net-kourier/releases/download/v"+kourierVersion+"/kourier.yaml")
-	if err := kourier.Run(); err != nil {
-		return fmt.Errorf("apply: %w", err)
+	for i := 0; i <= 2; {
```

Comment:
I don't understand this loop, it doesnt check any errors so it always runs the download three times, and always takes 15 seconds?

Reply:
Should probably have been something like this:
```go
        for i := 0; i <= 2; i++ {
		if err := kourier.Run(); err != nil {
			if i >=2 {
				return fmt.Errorf("apply: %w", err)
			}
			time.Sleep(5 * time.Second)
		} else {
			break
		}
	}
```

Reply:
you already have apimachinery in go.sum, so you might be better off using one of the [wait.Poll](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait) methods from there.

Reply:
I'm probably missing something obvious, but what would be the ConditionFunc in that scenario?

Reply:
Something like:

~~~
if err := wait.PollImmediate(1 * time.Second, 5 * time.Second, func() (bool, error) {
   return kourier.Run() == nil, nil
}; err != nil {
   return fmt.Errorf("wait: %w", lastErr)
}
~~~